core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -348,14 +357,15 @@ final class KafkaMetadataLog private (
           snapshotId.offset <= latestSnapshotId.offset &&
           log.maybeIncrementLogStartOffset(snapshotId.offset, LogStartOffsetIncrementReason.SnapshotGenerated) =>
             // Delete all segments that have a "last offset" less than the log start offset
-            log.deleteOldSegments()
+            val deletedSegments = log.deleteOldSegments()
             // Remove older snapshots from the snapshots cache
-            (true, forgetSnapshotsBefore(snapshotId))
+            val forgottenSnapshots = forgetSnapshotsBefore(snapshotId)
+            (deletedSegments != 0 || forgottenSnapshots.nonEmpty, forgottenSnapshots)
```

Comment:
Interesting, did we have a bug here previously? Looks like before we would always report `deleted=true` even if nothing was deleted. I wonder if we even use this return value 🤔

Reply:
The old code was probably correct in practice. I just wanted to make clearer that either one could be true for this function to return true.